### ISSUE TITLE
Incorrect attendee roster on refresh for one of the participants

### ISSUE THREAD

**Issue by Ajamuar** [aws/amazon-chime-sdk-js#2099]
### What happened and what did you expect to happen?

We are using `useAudioVideo` hook for our use cases. On loading for the first , `audioVideo?.realtimeSubscribeToAttendeeIdPresence(attendeeCallback)` calls the callback passed to it. But if we refresh the page, the callback from the same place is not called. Currently we are only capturing `Video` on one side and `None` on the other. When we change this to capture `VideoAndAudio` and `Audio` respectively, the issue is resolved. I am attaching a code snippet for your reference.

```tsx
import { useState, useEffect } from 'react';
import { useAudioVideo } from 'amazon-chime-sdk-component-library-react';
import { AudioVideoFacade } from 'amazon-chime-sdk-js';
import { MEDIAPIPELINE_BOT_EXTERNAL_ATTENDEE_ID } from '../../../constants';

const useAttendeeRoster: (maxParticipants: number) => Set<string> = (
  maxParticipants: number,
): Set<string> => {
  const [attendeeSet, setAttendeeSet] = useState<Set<string>>(new Set());
  const audioVideo: AudioVideoFacade = useAudioVideo();
  useEffect(() => {
    const attendeeCallback = (
      presentAttendeeId: string,
      present: boolean,
      externalUserId: string,
    ) => {
      // Do Something
    };
    audioVideo?.realtimeSubscribeToAttendeeIdPresence(attendeeCallback);
    return () => audioVideo?.realtimeUnsubscribeToAttendeeIdPresence(attendeeCallback);
  }, [audioVideo, maxParticipants]);
  return attendeeSet;
};

export default useAttendeeRoster;
```

### Have you reviewed our existing documentation?

- [X] GitHub [issues](https://github.com/aws/amazon-chime-sdk-js/issues)
- [X] FAQs [page](https://aws.github.io/amazon-chime-sdk-js/modules/faqs.html)
- [X] README [page](https://github.com/aws/amazon-chime-sdk-js#amazon-chime-sdk-for-javascript)
- [X] Feature-specific [guides](https://github.com/aws/amazon-chime-sdk-js#guides)

### Reproduction steps

1. Join a meeting using `meetingManager.join` as while capturing only `Video (2)` on one end and `None (0)` on the other.
2. Once the meeting starts on both end, refresh the page with `None (0)` capturing.
3. Check if callback occurs from the useAudioVideo() object.
```tsx

const audioVideo: AudioVideoFacade = useAudioVideo();
...
audioVideo?.realtimeSubscribeToAttendeeIdPresence(attendeeCallback);

```
5. attendeeCallback won't be called after the refresh.

### Amazon Chime SDK for JavaScript version

amazon-chime-sdk-js: "2.26.0" / amazon-chime-sdk-component-library-react: "2.13.0"

### What browsers are you seeing the problem on?

Google Chrome

### Browser version

99.0.4844.51

### Browser console logs

<redacted>

**Comment by michhyun1** [comment]
@Ajamuar thanks for bringing this up. I see that the SDK publishes the attendee presence event in this task: [ListenForVolumeIndicatorsTask](https://github.com/aws/amazon-chime-sdk-js/blob/6bf4f0244e7e77478691adad4d3cc6783dc1d181/src/task/ListenForVolumeIndicatorsTask.ts#L46). 

Could you double check - perhaps check the audioVideo is null when you refresh for some reason? Perhaps the subscribe is not getting called at all?

**Comment by michhyun1** [comment]
I also found this comment, not sure if increasing the `attendeePrescenceTimeout` would help: https://github.com/aws/amazon-chime-sdk-js/blob/6bf4f0244e7e77478691adad4d3cc6783dc1d181/src/audiovideocontroller/DefaultAudioVideoController.ts#L616-L629

It says that "No attendee presence event will be triggered if there is no audio connection." so I'd first question what happens to your audio connection when you refresh.

**Comment by Ajamuar** [comment]
@michhyun1 Our use case involves not having any audio input at all on either side. So far, we have realised that if we ask for audio resource, this doesn't happen. The above issue happens only when we don't demand audio from both sides. As for subscribe, we checked and it is being called.

**Comment by michhyun1** [comment]
@Ajamuar thanks for those details. I'll try to reproduce this on my side and I'll get back to you as soon as I can.

**Comment by michhyun1** [comment]
Actually, I am able to reproduce this. I just tried our [react components demo](https://github.com/aws-samples/amazon-chime-sdk/tree/main/apps/meeting). What I did was I changed this [line of code to allow me to join the meeting using only `None` (spectator) or `Video` mode only](https://github.com/aws-samples/amazon-chime-sdk/blob/6d528eee444966e71e7329b2958a2a19e2448e55/apps/meeting/src/containers/MeetingForm/index.tsx#L95-L96).

I joined from two different browsers, one as a spectator (`None`) and the other as only `Video`. The video client shows the roster (which uses the realtimeSubscribeToAttendeeIdPresence observer to populate).

The spectator, however, was not able to see the roster. The video client cannot see the spectator in the roster. However, if the video client turns on their video, both the spectator and the video client can see the video stream.

One thing I found - and the reason why the `realtimeSubscribeToAttendeeIdPresence` callback is only called when you join with `audio` - is because if you look in the JS SDK for when we actually publish that `realtimeAttendeeIdPresence` is [when we get a event from the signaling client](https://github.com/aws/amazon-chime-sdk-js/blob/646a9239ffeaef8956550fd53dd5838c9adb3e99/src/task/ListenForVolumeIndicatorsTask.ts#L43-L49) - and the event is an audio stream id. 

That means - technically - you need an audio stream to be "present" in a meeting. However, we should be sending a fake audio stream when you join with just `video` or just `none` so that they can still be "present" in the meeting. It seems to work fine with `Video`, but its not working for the spectator. I'll need to investigate this more.

As a mitigation, can you use `Video` and `Video` for now, but just disable the Video button for the spectator client?

**Comment by michhyun1** [comment]
Or if you want to use something with `Audio{andVideo}` - thats fine too

**Comment by michhyun1** [comment]
However, I should also mention a limitation with sending a "fake" audio stream is that in order to generate a "fake" audio stream we may still need to trigger device permissions dialogue on some browsers. We have to show the device permissions dialogue anyways for `Video`, so that's fine, but we currently don't for `None`.

This is the current limitation with this approach.

**Comment by Ajamuar** [comment]
@michhyun1 Thanks a lot for the response. We are going to go ahead and ask for audio to navigate around this problem for now. This would add another request for permission which we would like to avoid in future. It would be great if we are able to find another solution.

**Comment by michhyun1** [comment]
@Ajamuar There is one thing that we found in our React demo - if you join without audio and video, the realtimeSubscription for the presence works, and I see myself in the roster. However, if I enable WebAudio for that spectator when I join the meeting, I don't see the roster.

**Comment by mark-vanr** [comment]
Hi @michhyun1 you mentioned you've already been able to reproduce this issue but I ran into something similar a couple times over the last few days in my client app as well. 
I did join with audio and video permissions allowed, available and enabled; it may be that my underlying system did not have the streams free for a different reason. I joined the meeting but did not see the other attendees, which I confirmed out-of-band were all in the meeting. The other attendees also did not see me join the roster. I've added logs from a couple instances where it happened; figured it would be worth confirming if this appears to be the same issue or a different one.

First occurrence (info and verbose logs)
[info_me_all_alone.log](https://github.com/aws/amazon-chime-sdk-js/files/8390962/info_me_all_alone.log)
[verbose_me_all_alone.log](https://github.com/aws/amazon-chime-sdk-js/files/8390966/verbose_me_all_alone.log)
Second occurrence:
[info_alone_again.log](https://github.com/aws/amazon-chime-sdk-js/files/8390967/info_alone_again.log)

Experienced on Ubuntu; Chrome 100.0.4896.60; aws js-sdk v2.30.0

**Comment by simmkyu** [comment]
## Problem

In the [Chime SDK React Component Library Demo](https://github.com/aws-samples/amazon-chime-sdk/tree/main/apps/meeting), you don't see any attendee in the roster if you enable both **Join w/o Audio and Video** and **Enable Web Audio**.

<img width="263" alt="image" src="https://user-images.githubusercontent.com/36240708/161144789-2a5f210b-aeec-474f-a0d6-0923b07c41a0.png">

### What are **Join w/o Audio and Video** and **Web Audio**?

- **Join w/o Audio and Video**:  When enabled, the demo passes `DeviceLabels.None` to join a meeting without audio and video. The [Chime SDK for JavaScript](https://github.com/aws/amazon-chime-sdk-js) creates an empty audio stream, instead of using the microphone stream.
- **Web Audio**:  When enabled, the Chime SDK for JavaScript uses Web Audio nodes to process an incoming audio stream, such as applying [Amazon Voice Focus](https://aws.github.io/amazon-chime-sdk-js/modules/amazonvoice_focus.html) to the microphone stream.


Note that the demo works okay when you disable Web Audio.

⠀

## Root cause



The Chime SDK for JavaScript fails to route a fake audio stream

**Comment by simmkyu** [comment]
## Problem (Chrome browsers)

In the [Chime SDK React Component Library Demo](https://github.com/aws-samples/amazon-chime-sdk/tree/main/apps/meeting), you don't see any attendee in the roster if you enable **Join w/o Audio and Video** and **Enable Web Audio**. 

<img width="263" alt="image" src="https://user-images.githubusercontent.com/36240708/161144789-2a5f210b-aeec-474f-a0d6-0923b07c41a0.png">

### What are **Join w/o Audio and Video** and **Web Audio**?

- **Join w/o Audio and Video**:  When enabled, the demo passes DeviceLabels.None to join a meeting without audio and video. Instead of using the microphone stream, the [Chime SDK for JavaScript](https://github.com/aws/amazon-chime-sdk-js) creates an empty audio stream.
    ```js
    await meetingManager.join(meetingSessionConfiguration, {
      deviceLabels: DeviceLabels.None,
      enableWebAudio: true,
    });
    ```
- **Web Audio**:  When enabled, the Chime SDK for JavaScript uses Web Audio nodes to process an incoming audio stream, such as applying a noise suppressor [Amazon Voice Focus](https://aws.github.io/amazon-chime-sdk-js/modules/amazonvoice_focus.html) to the microphone stream.


Note that the demo works okay when you disable Web Audio.

⠀

## Root cause

When enabling **Join w/o Audio and Video** and **Enable Web Audio**, the Chime SDK for JavaScript fails to route an empty audio stream to a Web Audio input node.

<img width="888" alt="image" src="https://user-images.githubusercontent.com/36240708/161148999-1defa600-90ab-4b28-a8aa-9a5576fe97db.png">

⠀

## Workaround

- **Option 1:** Disable Web Audio when you join a meeting without audio. Web Audio has no impact because an empty audio stream does not require [Amazon Voice Focus and Echo Reduction](https://aws.github.io/amazon-chime-sdk-js/modules/amazonvoice_focus.html).
- **Option 2**: Force routing an empty audio stream to a Web Audio input node. To do so, you can apply 
    ```js
    import { useSelectAudioInputDevice } from 'amazon-chime-sdk-component-library-react';

    // After joining a meeting without audio, choose null to indicate a no-audio device explicitly.
    const selectAudioInputDevice = useSelectAudioInputDevice();
    useEffect(() => {
      selectAudioInputDevice(null);
    }, []);
    ```

⠀

## Next steps

- The Chime SDK team will include a fix in the upcoming release (v3).

**Comment by LeviSklaroff** [comment]
A bug fix for this issue was released in version 3.0.0 for amazon-chime-sdk-js #2144 please check it out.
